src/shillelagh/adapters/registry.py

```diff
@@ -8,7 +8,7 @@
 from collections import defaultdict
 from typing import Dict, List, Optional, Type, cast
 
-from pkg_resources import iter_entry_points
+from importlib_metadata import entry_points
```

Comment:
is this the right way to handle the imports where <3.10 and >= 3.10 are both supported?

Reply:
Yes, good point. That would indeed have the benefit of not requiring an install of the backport on 3.10+ and I don't see any downsides to it. I'll change it to the conditional way. Thanks